services/scrutinizer/scrutinizer-coverage.service.js

```diff
@@ -0,0 +1,157 @@
+'use strict'
+
+const Joi = require('joi')
+const { colorScale } = require('../color-formatters')
+const ScrutinizerBase = require('./scrutinizer-base')
+
+const schema = Joi.object({
+  default_branch: Joi.string().required(),
+  applications: Joi.object()
+    .pattern(
+      /^/,
+      Joi.object({
+        index: Joi.object({
+          _embedded: Joi.object({
+            project: Joi.object({
+              metric_values: Joi.object({
+                'scrutinizer.test_coverage': Joi.number().positive(),
+              }).required(),
+            }).required(),
+          }).required(),
+        }).required(),
+      })
+    )
+    .required(),
+}).required()
+
+// https://scrutinizer-ci.com/g/filp/whoops/code-structure/master/code-coverage
+// < 40% - red
+// 40-60% (inclusive) - yellow
+// > 60% brightgreen
+const scale = colorScale([40, 61], ['red', 'yellow', 'brightgreen'])
+
+class ScrutinizerCoverageBase extends ScrutinizerBase {
+  static get category() {
+    return 'coverage'
+  }
+
+  static get defaultBadgeData() {
+    return {
+      label: 'coverage',
+    }
+  }
+
+  static render({ coverage }) {
+    if (isNaN(coverage)) {
```

Comment:
Is this an error? Should it be checked in `transform()`?

Reply:
copy/paste from legacy implementation!

I actually don't think this a possible scenario anymore given what's already in place in the schema and `transform` so I'll update it accordingly

Reply:
Apologies, I was wrong (one of the downsides of working on this off and on for nearly a month). The current behavior of our badge is to display `unknown` when the specified repo/project doesn't have any coverage available. 

I've updated this behavior a bit by checking for coverage in the `transform` function, and throwing an error with a more explicit `coverage not found` error (open to any alternative error messages) on that scenario, removing this particular check in the `render` function. 

It is admittedly another change to the existing implementation, but I think an improvement.